storage/innobase/fts/fts0que.cc

```diff
@@ -3955,6 +3958,10 @@ fts_query_can_optimize(
 		return;
 	}
 
+	if (query->limit != ULONG_UNDEFINED) {
```

Comment:
Could you please explain this bit?

Reply:
`fts_query_can_optimize` enables one more memory optimization, the one implemented in 9f79ead0ac7. It solves the same issue of excess memory consumption by FTS query result as 2cd0ebf does. So, there is no point to enable both of them. This bit is unnecessary to fix the crash and can be removed of course.